client/src/document/ingredients/spec-section.tsx

```diff
@@ -17,6 +17,37 @@ export function SpecificationSection({
   }>;
   query: string;
 }) {
+  function TableRow(props) {
+    const spec = props.specification;
+
+    let shortTitle = "";
+    if (spec.title !== spec.shortTitle) {
+      shortTitle = `(${spec.shortTitle})`;
+    }
+
+    let anchor;
+    const hasAnchor = spec.bcdSpecificationURL.includes("#");
+    if (hasAnchor) {
+      anchor = (
+        <span>
+          {" "}
+          <br />{" "}
+          <small> # {`${spec.bcdSpecificationURL.split("#")[1]}`} </small>{" "}
+        </span>
+      );
+    }
+
+    return (
+      <tr key={spec.bcdSpecificationURL}>
+        <td>
+          <a href={spec.bcdSpecificationURL}>
+            {spec.title} {shortTitle} {anchor}
```

Comment:
I think should embrace JSX here and type this out as:
```suggestion
            {spec.title} {spec.title !== spec.shortTitle && `(${spec.shortTitle})`}<br/>
            {spec.bcdSpecificationURL.includes("#") && <small># {`${spec.bcdSpecificationURL.split("#")[1]}`}</small>}
```

Reply:
If you can express it all with JSX in "one line" you don't need to create a separate closure function. 
And that will solve the problem that you're setting `key={spec.bcdSpecificationURL}` "outside" the loop.